modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/crop_region.js

```diff
@@ -1,57 +1,59 @@
 AUI.add(
 	'liferay-crop-region',
 	function(A) {
-		var Lang = A.Lang;
-
-		var CropRegion = function() {
-		};
-
-		CropRegion.prototype = {
-			_getCropRegion: function(imagePreview, region) {
-				var instance = this;
-
-				var naturalSize = instance._getImgNaturalSize(imagePreview);
-
-				var scaleX = naturalSize.width / imagePreview.width();
-				var scaleY = naturalSize.height / imagePreview.height();
-
-				var regionHeight = region.height ? (region.height * scaleY) : naturalSize.height;
-				var regionWidth = region.width ? (region.width * scaleX) : naturalSize.width;
-
-				var regionX = region.x ? Math.max(region.x * scaleX, 0) : 0;
-				var regionY = region.y ? Math.max(region.y * scaleY, 0) : 0;
-
-				return {
-					height: regionHeight,
-					width: regionWidth,
-					x: regionX,
-					y: regionY
-				};
-			},
-
-			_getImgNaturalSize: function(img) {
-				var instance = this;
-
-				var imageHeight = img.get('naturalHeight');
-				var imageWidth = img.get('naturalWidth');
-
-				if (Lang.isUndefined(imageHeight) || Lang.isUndefined(imageWidth)) {
-					var tmp = new Image();
-
-					tmp.src = img.attr('src');
-
-					imageHeight = tmp.height;
-					imageWidth = tmp.width;
+		if (!Liferay.Util.getCropRegion) {
```

Comment:
Are we sure we want to do this? This seems risky for backward compatibility. The original plan was to deprecate old utility and leave it as it is.

Reply:
This was the way Chema told me to do it for the Address module.

Reply:
This should be fine and should be more robust. If the module is loaded after the new utility, it won't replace the feature, so it should be safer to remove down the road because everyone would be using the new methods.